windows/Dockerfile

```diff
@@ -0,0 +1,51 @@
+FROM mcr.microsoft.com/windows/servercore:1809 AS caddy
+
+SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]
+
+# https://github.com/caddyserver/dist/commits
+ENV CADDY_DIST_COMMIT 80870b227ded910971ecace4a0c136bf0ef46342
+
+RUN mkdir /config \
+    mkdir /data \
+    mkdir /caddy \
+    mkdir /usr/share/caddy; \
+    Invoke-WebRequest \
+        -Uri "https://github.com/caddyserver/dist/raw/$env:CADDY_DIST_COMMIT/config/Caddyfile" \
+        -OutFile "/caddy/Caddyfile"; \
+    Invoke-WebRequest \
+        -Uri "https://github.com/caddyserver/dist/raw/$env:CADDY_DIST_COMMIT/welcome/index.html" \
+        -OutFile "/usr/share/caddy/index.html"
+
+# https://github.com/caddyserver/caddy/releases
+ENV CADDY_VERSION v2.0.0-rc.3
+
+RUN Invoke-WebRequest \
+        -Uri "https://github.com/caddyserver/caddy/releases/download/v2.0.0-rc.3/caddy_2.0.0-rc.3_windows_amd64.zip" \
+        -OutFile "/caddy.zip"; \
+    Expand-Archive -Path "/caddy.zip" -DestinationPath "/" -Force; \
+    Remove-Item "/caddy.zip" -Force
+
+# See https://caddyserver.com/docs/conventions#file-locations for details
+ENV XDG_CONFIG_HOME=/config
+ENV XDG_DATA_HOME=/data
+
+VOLUME /config
+VOLUME /data
+
+LABEL org.opencontainers.image.version=v2.0.0-rc.3
+LABEL org.opencontainers.image.title=Caddy
+LABEL org.opencontainers.image.description="a powerful, enterprise-ready, open source web server with automatic HTTPS written in Go"
+LABEL org.opencontainers.image.url=https://caddyserver.com
+LABEL org.opencontainers.image.documentation=https://caddyserver.com/docs
+LABEL org.opencontainers.image.vendor="Light Code Labs"
+LABEL org.opencontainers.image.licenses=Apache-2.0
+LABEL org.opencontainers.image.source="https://github.com/caddyserver/caddy-docker"
+
+EXPOSE 80
+EXPOSE 443
+EXPOSE 2019
+
+# Make sure it runs and reports its version
+RUN ["caddy", "version"]
+
+CMD ["caddy", "run", "--config", "/caddy/Caddyfile", "--adapter", "caddyfile"]
```

Comment:
Looks like the default Caddyfile is in a different path than on the linux variants. Kinda less fun to document that. Can we unify that?

Reply:
Yeah, I waffled on that for a bit... We can definitely put it in `/etc/caddy/`. It translates to `C:\etc\caddy\` in Windows, but `/` works fine 😉